### PR TITLE
container: bring back support for lxc 2.x under a feature flag

### PR DESCRIPTION
**Pull request by morphis**
Some of our users (Ubports) are still using an older version of liblxc
(2.x as in Ubuntu 16.04) and we still want to support them. Support for
LXC 2.x is now available again and auto detected at configuration time.

### CODE REVIEW

**Comment by morphis** [comment]
@mariogrip Let me know if this helps you.

**Comment by mariogrip** [comment]
@morphis Awesome! will give it a test, will also cleanup ubports branch (and upstream it)

**Comment by morphis** [comment]
@mariogrip Let me know how testing goes, then I will merge this in.

File: src/anbox/container/lxc_container.cpp [hunk 1]
```diff
@@ -278,25 +307,28 @@ void LxcContainer::start(const Configuration &configuration) {
...
 
-  set_config_item("lxc.log.level", "0");
+  set_config_item(lxc_config_log_level_key, "0");
   const auto log_path = SystemConfiguration::instance().log_dir();
-  set_config_item("lxc.log.file", utils::string_format("%s/container.log", log_path).c_str());
+  set_config_item(lxc_config_log_file_key, utils::string_format("%s/container.log", log_path).c_str());
+
+#ifndef ENABLE_LXC2_SUPPORT
+    // Dump the console output to disk to have a chance to debug early boot problems
+    set_config_item("lxc.console.logfile", utils::string_format("%s/console.log", log_path).c_str());
```
Comment (zhsj) [inline src/anbox/container/lxc_container.cpp]: lxc2 has `lxc.console.logfile`, but no `lxc.console.rotate`

Comment (mariogrip) [reply src/anbox/container/lxc_container.cpp]: Log seems to fail on start up on lxc2 for some reason, even without rotate.

Comment (morphis) [reply src/anbox/container/lxc_container.cpp]: @mariogrip `lxc.console.rotate` doesn't exist on LXC 2.x for sure. 

However I wouldn't recommend enabling `lxc.console.logfile` for LXC 2.x if not absolutely necessary as the implementation in LXC is the best and was improved to be based on a much better performing ring buffer in 3.x. Check https://discuss.linuxcontainers.org/t/lxc-3-0-0-has-been-released/1449

**Comment by morphis** [comment]
@mariogrip Does this otherwise work for you so we can merge it?

**Comment by mariogrip** [comment]
@morphis Jep, this works :+1:  Thanks :)

**Comment by morphis** [comment]
@mariogrip Great, lets merge this then.

**Comment by mariogrip** [comment]
@morphis Jep go ahead :)

**Pull request merged**